src/Features/Core/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -87,7 +87,10 @@ public State(TService service, SemanticDocument document)
                     return false;
                 }
 
-                this.IsConstant = this.Document.SemanticModel.GetConstantValue(this.Expression, cancellationToken).HasValue;
+                // Interpolated strings are not constants themselves, even if they were composed entirely from constants.
+                this.IsConstant = _service.IsInterpolatedStringExpression(this.Expression)
+                    ? false
+                    : this.Document.SemanticModel.GetConstantValue(this.Expression, cancellationToken).HasValue;
```

Comment:
I thought about it too, but from the current implementation, I assumed that `GetConstantValue` returns the `constant` part, if any, in a given expression. For `$"{DateTime.Now.ToString()}Text{args[0]}"` , it returns `"Text"` as the constant value. Not sure if that is the intended behavior or a bug.